InAppUtils/InAppUtils.m

```diff
@@ -220,12 +220,18 @@ - (void)productsRequest:(SKProductsRequest *)request
 }
 
 // SKProductsRequestDelegate network error
-- (void)request:(SKRequest *)request didFailWithError:(NSError *)error{
+- (void)request:(SKRequest *)request didFailWithError:(NSError *)error
+{
     NSString *key = RCTKeyForInstance(request);
     RCTResponseSenderBlock callback = _callbacks[key];
-    if(callback) {
+    if (callback) {
+        // Ensure error.userData can be converted to JSON without error.
+        // This will remove any NSURL from the error.userData.
+        error = RCTErrorClean(error);
```

Comment:
agreed. let's get a PR upstream. once it's merged upstream, we can add a fallback here. If it's not merged upstream in x time, then let's ship here.

that works for you?